core/google/cloud/client.py

```diff
@@ -126,6 +128,14 @@ def __init__(self, credentials=None, http=None):
             credentials, self.SCOPE)
         self._http_internal = http
 
+    def __getstate__(self):
+        """Explicitly state that clients are not pickleable."""
+
```

Comment:
That used to be required by PEP 8, and I got used to that before PEP 257 became a thing and forbade them. (I actually had to check PEP 257 now to make sure.)